swarm/hash/multihash.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package swarmhash
```

Comment:
lets not use swarm hash cos it is something else already

Reply:
But the swarmhash will be here in the end, no?

Reply:
Sure, we can call it something else. But `hash` is not good, because that's the name of a package already in use in the stack. And `multihash` is misleading because the package will / is indended to cover all hashing functionality in swarm.

Any other good name suggestions? `hashing` ? `hasher`?

Reply:
multihash is exactly the name since it concerns a self describing hash, hence it is a multihash. or am i seeing things the wrong way?

Reply:
> And multihash is misleading because the package will / is indended to cover all hashing functionality in swarm.

Reply:
I vote "multihash"

Reply:
@fjl you would vote `swarm/multihash` even if it does more than just treating multihashes?